xla/backends/gpu/runtime/dynamic_slice_thunk_test.cc

```diff
@@ -1537,4 +1547,177 @@ TEST(DynamicSliceThunkTest, SlicedOperandsSameBufferGemm) {
+  int64_t out_length = sizeof(float) * 1 * 1;
+
+  // Step 1:
+  // Prepare embedded and address computation thunks.
+
+  // Preparing buffer allocation slices for thunk creations.
+  std::vector<std::unique_ptr<BufferAllocation>> fake_allocations(4);
+
+  fake_allocations.push_back(
+      std::make_unique<BufferAllocation>(/*index=*/0, rhs_length, /*color=*/0));
+  BufferAllocation::Slice slice_lhs_fake(fake_allocations.back().get(), 0,
+                                         rhs_length);
+
+  BufferAllocation alloc_lhs(/*index=*/0, lhs_length, /*color=*/0);
+  BufferAllocation::Slice slice_lhs(&alloc_lhs, 0, lhs_length);
+
+  fake_allocations.push_back(
+      std::make_unique<BufferAllocation>(/*index=*/1, rhs_length, /*color=*/0));
+  BufferAllocation::Slice slice_rhs(fake_allocations.back().get(), 0,
+                                    rhs_length);
+
+  fake_allocations.push_back(
+      std::make_unique<BufferAllocation>(/*index=*/2, out_length, /*color=*/0));
+  BufferAllocation::Slice slice_out(fake_allocations.back().get(), 0,
+                                    out_length);
+
+  fake_allocations.push_back(std::make_unique<BufferAllocation>(
+      /*index=*/3, 1024 * 1024, /*color=*/0));
+  BufferAllocation::Slice slice_workspace(fake_allocations.back().get(), 0,
+                                          1024 * 1024);
+
+  // Preparing config for GEMM thunk.
+  auto config = GemmConfig::For(
+      ShapeUtil::MakeShape(PrimitiveType::F32, {1, 4}), {}, {1},
+      ShapeUtil::MakeShape(PrimitiveType::F32, {4, 1}), {}, {0},
+      ShapeUtil::MakeShape(PrimitiveType::F32, {1, 1}), 1.0, 0.0, 0.0,
+      PrecisionConfig::ALG_UNSET, std::nullopt,
+      se::blas::kDefaultComputePrecision, false, false,
+      executor->GetDeviceDescription().gpu_compute_capability());
+  ASSERT_TRUE(config.ok());
+
+  // Creating embedded GEMM thunk.
+  ThunkSequence seq;
+  seq.emplace_back(std::make_unique<GemmThunk>(
+      Thunk::ThunkInfo(), config.value(), slice_lhs_fake, slice_rhs, slice_out,
+      slice_workspace, /*deterministic=*/true));
+
+  // Wrapping address computation thunk around the GEMM thunk.
+  std::vector<DynamicSliceThunk::Offset> lhs_offsets{offset_module, 0l};
+  DynamicSliceThunk thunk(
+      /*thunk_info=*/Thunk::ThunkInfo(),
+      /*embedded_thunk=*/std::make_unique<ThunkSequence>(std::move(seq)),
+      /*arguments=*/{slice_lhs, slice_rhs, slice_out, slice_workspace},
+      /*fake_allocations=*/std::move(fake_allocations),
+      /*offsets=*/{lhs_offsets, std::nullopt, std::nullopt, std::nullopt},
+      /*orig_shapes=*/
+      {ShapeUtil::MakeShape(PrimitiveType::F32, {2, 4}), std::nullopt,
+       std::nullopt, std::nullopt},
+      /*sliced_shapes=*/
+      {ShapeUtil::MakeShape(PrimitiveType::F32, {1, 4}), std::nullopt,
+       std::nullopt, std::nullopt},
+      /*offset_byte_sizes=*/
+      {sizeof(int64_t), std::nullopt, std::nullopt, std::nullopt},
+      /*temp_modules=*/std::move(fake_modules),
```

Comment:
I guess this won't match the name, right? Thought we got rid of `temp_modules` :)